src/app/components/passwordToggleButton.component.ts

```diff
@@ -0,0 +1,28 @@
+import { PasswordState } from '../components';
+
+export class PasswordToggleButton{
+    public create(passwordInput: HTMLInputElement) : HTMLButtonElement {
+        const showButton = document.createElement('button');
+        showButton.textContent = 'Show';
+
+        showButton.addEventListener('click', (event) => {
+          event.preventDefault();
```

Comment:
Please, check if we really need to prevent default event execution ?

Reply:
yes, we really need to use prevent default